src/Helper/Installation.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Castor\Helper;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\DependencyInjection\Attribute\Exclude;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\Cache\CacheInterface;
+
+/**
+ * @internal
+ */
+#[Exclude]
+final class Installation
+{
+    private ?Architecture $architecture = null;
+    private ?InstallationMethod $method = null;
+    private ?string $path = null;
```

Comment:
i would remove the nullable for those fields, if it's only to handle the "initialized" state, PHP already does that so no need to repeat that. AFAIK detect method ensure there is always a value